src/setup-beam.js

```diff
@@ -278,7 +266,10 @@ async function getRebar3Versions() {
 }
 
 function getVersionFromSpec(spec, versions) {
-  if (versions.includes(spec)) {
+  if (
+    versions.includes(spec) ||
+    core.getInput('version-type', { required: false }) === 'strict'
```

Comment:
I'm tentatively adding this here. I ran into version-related trouble while implementing Windows and feel since this has been reported as an issue before it might just be time to implement it. Adding it to this pull request since a bug, made evident, seems impossible to overcome with such a mechanism.